HPCC-13236 Make query list checking status on all roxie nodes optional

afishbeck opened the pull request:
Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

afishbeck committed:
HPCC-13236 Make query list checking status on all roxie nodes optional

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck commented:
Makes checking status on all roxie nodes optional from both command line and EclWatch.

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
At a quick reading, this seems like it would change the behaviour for anyone using the old interface/not supplying a value for the new parameter to the SOAP service?

afishbeck commented:
That's correct.  I can fix the default behaviors.  Only really helps if the default for EclWatch is to check only one roxie node.

Or, I could try making the EclWatch default check-all-nodes, but use  a cookie to remember if they've changed it and from then on for that user it would be faster.  People would have to be educated in order to benefit.

richardkchapman commented:
It's ok for eclwatch and our other clients to be changed to default to checking one node, but you should not change the default behaviour of the SOAP interface (was the point I was trying to make)

afishbeck committed:
HPCC-13236 Updated based on review

Make QueryList soap calls default to checking all nodes for status.

Also make improve how opion parameters are gathered from the
options drop down form.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck commented:
@richardkchapman Updated.

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

miguelvazq commented on esp/src/eclwatch/templates/QuerySetQueryWidget.html:
```diff
@@ -33,6 +33,22 @@
+                        <div data-dojo-type="dijit.TooltipDialog">
+                            <div id="${id}OptionsForm" style="width: 530px;" onsubmit="return false;" data-dojo-type="dijit.form.Form">
+                                <div data-dojo-type="hpcc.TableContainer">
+			                        <select id="${id}CheckAllNodes" title="${i18n.Status}:" name="CheckAllNodes" colspan="2" data-dojo-type="dijit.form.Select">
```
Minor comment on indentation here.

miguelvazq commented:
Otherwise, looks good.

miguelvazq commented on esp/src/eclwatch/templates/QuerySetQueryWidget.html:
```diff
@@ -33,6 +33,22 @@
                             <option value="0">${i18n.NotActive}</option>
                         </select>
                     </div>
```
I would add <span data-dojo-type="dijit.ToolbarSeparator"></span> here to separate the controls on the toolbar.

afishbeck committed:
HPCC-13236 Fix whitepace issue and add toolbar separator

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck commented:
@miguelvazq updated

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

miguelvazq commented:
Looks good.
